File: python/triangle/triangle.py
def equilateral(sides):
    a, b, c = sides
    return valid(sides) and a == b and b == c


def isosceles(sides):
    a, b, c = sides
    return valid(sides) and (a == b or b == c or a == c)


def scalene(sides):
    a, b, c = sides
    return valid(sides) and a != b and b != c and a != c


def valid(sides):
    a, b, c = sides
    return all(side > 0 for side in sides) and a + b > c and a + c > b and b + c > a

File: julia/triangle/triangle.jl
is_valid(a, b, c) = all([side > 0 for side in [a, b, c]]) && 
    a + b ≥ c && c + b ≥ a && a + c ≥ b

function is_equilateral(sides)
    is_valid(sides...) || return false
    a, b, c = sides
    a == b == c
end

function is_isosceles(sides)
    is_valid(sides...) || return false
    a, b, c = sides
    a == b || b == c || c == a
end

is_scalene(sides) = is_valid(sides...) && !(is_isosceles(sides) || is_equilateral(sides))
